src/services/authorization.rs

```diff
@@ -1,199 +1,142 @@
-
-        let mut mock_repository = MockRepository::new();
-        mock_repository
-            .expect_get_compact()
-            .with(predicate::eq(test_user_id))
-            .times(1)
-            .returning(move |_| {
-                Ok(UserCompact {
-                    user_id: test_user_id,
-                    username: "non_admin_user".to_string(),
-                    administrator: false,
-                })
-            });
-
-        let service = Service::new(Arc::new(Box::new(mock_repository)));
-        assert_eq!(
-            service.authorize(ACTION::DeleteCategory, Some(test_user_id)).await,
-            Err(ServiceError::Unauthorized)
-        );
+        let mut enforcer = Enforcer::new(model, ()).await.expect("Error creating the enforcer");
+
+        enforcer.add_policies(policy).await.expect("Error loading the policy");
+
+        let enforcer = Arc::new(RwLock::new(enforcer));
+
+        Self { enforcer }
     }
+}
+#[allow(dead_code)]
+struct CasbinConfiguration {
+    model: String,
+    policy: Vec<Vec<std::string::String>>,
+}
 
-    #[tokio::test]
-    async fn an_admin_user_should_be_able_to_delete_categories() {
-        let test_user_id = 6;
-
-        let mut mock_repository = MockRepository::new();
-        mock_repository
-            .expect_get_compact()
-            .with(predicate::eq(test_user_id))
-            .times(1)
-            .returning(move |_| {
-                Ok(UserCompact {
-                    user_id: test_user_id,
-                    username: "admin_user".to_string(),
-                    administrator: true,
-                })
-            });
-
-        let service = Service::new(Arc::new(Box::new(mock_repository)));
-        assert_eq!(service.authorize(ACTION::DeleteCategory, Some(test_user_id)).await, Ok(()));
+impl CasbinConfiguration {
+    pub fn new() -> Self {
+        CasbinConfiguration {
+            model: String::from(
+                "
+                [request_definition]
+                r = role, action
+                
+                [policy_definition]
+                p = role, action
+                
+                [policy_effect]
+                e = some(where (p.eft == allow))
+                
+                [matchers]
+                m = r.role == p.role && r.action == p.action
+            ",
+            ),
+            policy: vec![
+                vec!["admin".to_string(), "AddCategory".to_string()],
+                vec!["admin".to_string(), "DeleteCategory".to_string()],
+                vec!["admin".to_string(), "GetSettings".to_string()],
+                vec!["admin".to_string(), "GetSettingsSecret".to_string()],
+                vec!["admin".to_string(), "AddTag".to_string()],
+                vec!["admin".to_string(), "DeleteTag".to_string()],
+                vec!["admin".to_string(), "DeleteTorrent".to_string()],
+                vec!["admin".to_string(), "BanUser".to_string()],
+            ],
+        }
```

Comment:
@mario-nt I also suggested in my previous review to use here a literal string too:

```rust
fn main() {
    let multiline_str = r#"
admin, AddCategory
admin, DeleteCategory
admin, GetSettings
admin, GetSettingsSecret
admin, AddTag
admin, DeleteTag
admin, DeleteTorrent
admin, BanUser
"#;

    let policy: Vec<Vec<String>> = multiline_str
        .lines()
        .filter(|line| !line.trim().is_empty())
        .map(|line| {
            line.split(',')
                .map(|s| s.trim().to_owned())
                .collect::<Vec<String>>()
        })
        .collect();

    println!("{:?}", policy);
}
```

I think it's clearer if we use a string in both cases. Besides, we will allow in the future to inject the polices via unstable config value as @da2ce7 suggested, so we will have to do it anyway.